ONSdigital/eq-author-app
# Ear 348 deleted page crash

> ### BEFORE MAKING YOUR PR
>
> Please ensure:
>
> - There are no linting errors, all tests must pass
> - PR is named after JIRA ticket number e.g. EAR-###
> - **Accesibility** checks are completed:
>   - Elements have discernible and consistent focus states
>   - Elements can be navigated to and used by just a keyboard
>   - Elements and text can be read out by a screen reader, and the descriptions are understandable
> - Your feature / bug fix works across **GCP** and **AWS** (where appropriate)
>   - Are modifications to eq-publisher-v3 required?
>   - Are modifications to the Firestore data source required?

---

### What is the context of this PR?

> When directed to a page or section that is deleted the app crashes (gives a white screen error).
> E.g. If you add a new page and delete it and try going back to it with the url you will get an error because you're going back  to a delete page which is now null.
> Redirected the user to go to the first page/ section of the questionnaire instead of the deleted question/ page design
> Used RedirectRoute component to take the user from the url entered to the first page or section of the questionnaire depending whether or not they go to a deleted page or section.
> added additional checks to fix some type error mainly on section code to make sure a section exist before checking its properties. 

### How to review

- run author...
- make a questionnaire (business or social)
- create a page and add contents in it.
- copy the url and then delete the page.
- paste the url of the delete page and see if you get a error.
- if you're redirected back to the first page of the questionnaire then its working.
- same with section.
